<a href="https://colab.research.google.com/github/BatsaikhanTuguldur/KousenProjects/blob/main/TITANIC_4J38_%E3%83%88%E3%83%BC%E3%82%B4%E3%83%BC.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

情報・ソフトウェア系  
４Jー３８番号  
トゥグルデレバトサイハン


In [ ]:
import seaborn as sns
import pandas as pd

titanic = sns.load_dataset('titanic')
titanic.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 15 columns):
 #   Column       Non-Null Count  Dtype   
---  ------       --------------  -----   
 0   survived     891 non-null    int64   
 1   pclass       891 non-null    int64   
 2   sex          891 non-null    object  
 3   age          714 non-null    float64 
 4   sibsp        891 non-null    int64   
 5   parch        891 non-null    int64   
 6   fare         891 non-null    float64 
 7   embarked     889 non-null    object  
 8   class        891 non-null    category
 9   who          891 non-null    object  
 10  adult_male   891 non-null    bool    
 11  deck         203 non-null    category
 12  embark_town  889 non-null    object  
 13  alive        891 non-null    object  
 14  alone        891 non-null    bool    
dtypes: bool(2), category(2), float64(2), int64(4), object(5)
memory usage: 80.6+ KB


In [ ]:
titanic.head()

survived  pclass     sex   age  ...  deck  embark_town  alive  alone
0         0       3    male  22.0  ...   NaN  Southampton     no  False
1         1       1  female  38.0  ...     C    Cherbourg    yes  False
2         1       3  female  26.0  ...   NaN  Southampton    yes   True
3         1       1  female  35.0  ...     C  Southampton    yes  False
4         0       3    male  35.0  ...   NaN  Southampton     no   True

[5 rows x 15 columns]

女性，男性の生存率はそれぞれ幾つでしょうか？


In [ ]:
#survival rate by sex
df_sex = titanic.groupby("sex").describe()

female_survival_rate = df_sex.iloc[0, 1]
male_survival_rate = df_sex.iloc[1, 1]

print("survival rate for women on Titanic is ", 100*female_survival_rate, "%")
print("survival rate for men on Titanic is", 100*male_survival_rate, "%")

survival rate for women on Titanic is  74.20382165605095 %
survival rate for men on Titanic is 18.890814558058924 %


survival rate for women on Titanic is  74.20382165605095 %  
survival rate for men on Titanic is 18.890814558058924 %

1等級，3等級のチケットを持っていた乗客の生存率はそれぞれ幾つでしょうか？

In [ ]:
#survival rate by class
df_pclass = titanic.groupby("pclass").describe()
#print(df_pclass)

pclass1 = df_pclass.iloc[0, 1]
pclass3 = df_pclass.iloc[2, 1]

print("If you had a 1st class ticket on titanic your survival rate would have been: " , 100*pclass1, "%")
print("If you had a 3rd class ticket on titanic your survival rate would have been: " , 100*pclass3, "%")


If you had a 1st class ticket on titanic your survival rate would have been:  62.96296296296296 %
If you had a 3rd class ticket on titanic your survival rate would have been:  24.236252545824847 %


If you had a 1st class ticket on titanic your survival rate would have been: 62.96296296296296 %  
If you had a 3rd class ticket on titanic your survival rate would have been:  24.236252545824847 %

使用したデータセットに含まれる10歳未満の乗客の数は幾つでしょうか？  
使用したデータセットにおいて，10歳未満の乗客の生存率は幾つでしょうか？


In [ ]:
#survival rate by kids under the age of 10
kid_num = len(titanic[titanic["age"]<10])
print("There were", kid_num, "kids under the age of 10 on Titanic")

kid_survival_rate = titanic[titanic["age"]<10]["survived"].mean()
print("If you were a kid under the age of 10 your survival rate would have been: ", 100*kid_survival_rate, "%")

There were 62 kids under the age of 10 on Titanic
If you were a kid under the age of 10 your survival rate would have been:  61.29032258064516 %


There were 62 kids under the age of 10 on Titanic  
If you were a kid under the age of 10 your survival rate would have been:  61.29032258064516 %

その他，自分で解析をしてみましょう．（新しい基準で解析を行ったり，データの可視化をしてみたり，検定を行ったりなど）  
survival rate by sex in classes


In [ ]:
#survival rate by sex from classes
titanic[['survived', 'sex', 'pclass']].groupby(['pclass', 'sex']).mean()


survived
pclass sex             
1      female  0.968085
       male    0.368852
2      female  0.921053
       male    0.157407
3      female  0.500000
       male    0.135447

pclass  sex  
   1   female	  96.8085%  
       male   	36.8852%  
	 2   female	  92.1053%  
       male	    15.7407%  
	 3   female	  50.0000%  
       male	    13.5447%  

survival rate by age


In [ ]:
age_labels = ['0-10', '11-20', '21-30', '31-40', '41-50', '51-60', '61-70', '71-80']
titanic['age_group'] = pd.cut(titanic.age, range(0, 81, 10), right=False, labels=age_labels)
age_survival_rate = titanic.groupby(['age_group']).survived.mean()
print(age_survival_rate)

age_group
0-10     0.612903
11-20    0.401961
21-30    0.350000
31-40    0.437126
41-50    0.382022
51-60    0.416667
61-70    0.315789
71-80    0.000000
Name: survived, dtype: float64


age_group  survival probability  
0-10     0.612903  
11-20    0.401961  
21-30    0.350000  
31-40    0.437126  
41-50    0.382022  
51-60    0.416667  
61-70    0.315789  


In [ ]:
data1 = titanic
for dataset in data1:
    dataset['age'] = dataset['age'].astype(int)
    dataset.loc[dataset['age'] <= 14, 'age'] = 0
    dataset.loc[(dataset['age'] > 14) & (dataset['age'] <= 20), 'age'] = 1
    dataset.loc[dataset['age'] > 20, 'age'] = 2

TypeError: ignored